mysql-persistence/src/main/java/com/netflix/conductor/dao/mysql/MySQLBaseDAO.java

```diff
@@ -20,8 +17,22 @@
 import java.util.Arrays;
 import java.util.List;
 import java.util.function.Consumer;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import static com.mysql.cj.exceptions.MysqlErrorNumbers.ER_LOCK_DEADLOCK;
+import static com.netflix.conductor.core.execution.ApplicationException.Code.BACKEND_ERROR;
+import static com.netflix.conductor.core.execution.ApplicationException.Code.CONFLICT;
+import static com.netflix.conductor.core.execution.ApplicationException.Code.INTERNAL_ERROR;
+import static java.lang.Integer.parseInt;
+import static java.lang.System.getProperty;
 
 public abstract class MySQLBaseDAO {
+    private static final String MAX_RETRY_ON_DEADLOCK_PROPERTY_NAME = "conductor.mysql.deadlock.retry.max";
```

Comment:
I did that at first as well as attempt to allowed it to be configured via the properties files.
Then I realised the way it works is that I have to inject the component. 
I tried that but then not sue we want that coupling.
(An opposite approach to gathering all config into one file, if the config only affect logic in certain class it should just be there and be private, the var name is simply to aid the documentation.)
The reason this props is private is because I think in the end this is an implementation detail.
Let me know if you still prefer it to be in MySqlConfiguration.